src/base/QXmppRegisterIq.cpp

```diff
@@ -272,7 +294,15 @@ void QXmppRegisterIq::parseElementFromChild(const QDomElement &element)
     d->username = queryElement.firstChildElement(QStringLiteral("username")).text();
     d->password = queryElement.firstChildElement(QStringLiteral("password")).text();
     d->email = queryElement.firstChildElement(QStringLiteral("email")).text();
-    d->form.parse(queryElement.firstChildElement(QStringLiteral("x")));
+
+    QDomElement xElement = queryElement.firstChildElement(QStringLiteral("x"));
+    if (!xElement.isNull()) {
+        if (xElement.namespaceURI() == ns_data)
+            d->form.parse(xElement);
+        else if (xElement.namespaceURI() == ns_oob)
+            d->outOfBandUrl = xElement.firstChildElement(QStringLiteral("url")).text();
+    }
```

Comment:
This does not work with a form and an oob URL. Iterate over the child elements and parse both.